### PR TITLE
Fix the autodeployment of the documentation

### PR DESCRIPTION
**Pull request by alessandropellegrini**
This pull requests interacts with ROOT-Sim/ci-actions@3adadf1aed204ea7647a45163fb71b5617010d23 to deploy the documentation for the `master` and `develop` branches correctly to the GitHub Pages website.

### CODE REVIEW

File: scripts/bump_version.py [hunk 1]
```diff
@@ -51,7 +52,7 @@
     cmakelists = f.read()
 
 cmakelists, sub_cnt = re.subn(r"PROJECT_VERSION .*\)",
-                               f"PROJECT_VERSION {major}.{minor}.{hotfix}{new_tag})",
+                               f"PROJECT_VERSION v{major}.{minor}.{hotfix}{new_tag})",
```
Comment (Piccions) [inline scripts/bump_version.py:55]: See my previous comment

File: scripts/bump_version.py [hunk 2]
```diff
@@ -61,4 +62,4 @@
 with open(build_path, "w") as f:
     f.write(cmakelists)
 
-print(f"File modified successfully, version bumped to {major}.{minor}.{hotfix}{new_tag}")
+print(f"File modified successfully, version bumped to v{major}.{minor}.{hotfix}{new_tag}")
```
Comment (Piccions) [inline scripts/bump_version.py:65]: See my previous comment

File: test/tests/visibility/visibility_weak.c [hunk 3]
```diff
@@ -11,5 +11,5 @@
 
 int main(void)
 {
-	return strcmp(core_version, "3.0.0-alpha.4");
+	return strcmp(core_version, "v3.0.0-alpha.5");
```
Comment (Piccions) [inline test/tests/visibility/visibility_weak.c:14]: See my previous comment

File: scripts/bump_version.py [hunk 4]
```diff
@@ -11,36 +11,37 @@
...
-    tag = tag.split(".")[0]
-    new_tag = f"{tag}.{tag_version + 1}"
+    new_tag = tag.split(".")[0]
+    new_tag = f"{new_tag}.{tag_version + 1}"
 
 branch_type = branch_name.split("-", 1)[0]
 
 print(f"The current branch {branch_name} is of type {branch_type}")
-print(f"Current version is {major}.{minor}.{hotfix}{new_tag}")
+print(f"Current version is v{major}.{minor}.{hotfix}{tag}")
```
Comment (Piccions) [inline scripts/bump_version.py:27]: See my previous comment

**Review by Piccions — COMMENTED** [review]
Thanks for the work on the website, I only opened a nit-picky discussion on the format of version string to use in the codebase.

**Comment by alessandropellegrini** [comment]
Thanks for pointing to the nits!

I understand there is a bit of confusion between version number and release number, and there is no actual agreement among those managing the two aspects. As you pointed out, the version number shall not pick the `v` prefix, although some semver libraries [explicitly handle it](https://github.com/asamuzaK/semverParser/blob/master/modules/semver.js#L21), although emphasizing that it's not strict versioning. Interestingly, also semver is [using the `v` prefix in their releases](https://github.com/semver/semver/tags).

From the release point of view, since we are using GitHub officially, I emphasize that also GitHub [encourages](https://github.com/ROOT-Sim/core/releases/new) using the `v` prefix.

Given all this, I agree on removing the `v` prefix from the version number but keeping it in tags and releases.

On the `PROJECT_VERSION` point, I emphasize that it's only _partially_ supporting semver, as [it is not supporting suffixes](https://gitlab.kitware.com/cmake/cmake/-/issues/16716), which are part of the strict semver specification. Considering that we will likely have in the future additional `alpha`, `beta`, and `rc` versions until they fix their support for a full semver support, we cannot avoid using the [current workaround](https://github.com/ROOT-Sim/core/blob/dfa89e4fe58f93e139144e7f79c64891b3a50e75/CMakeLists.txt#L6).

To summarize, I would like to:
* stick using `set(PROJECT_VERSION XXXXXXXX)` in `CMakeLists.txt`
* remove the `v` from the version number
* keep `v` in tags
* readapt `bump_version.py` to make it consistent with the changes.

Let me know if you agree on all this, and I'll update the PR accordingly.

**Comment by Piccions** [comment]
I completely agree with what you said and I approve your proposed solution and changes. Thanks for your work!

**Review by Piccions — APPROVED** [review]

**Review by AdrianoPi — APPROVED** [review]

**Comment by Piccions** [comment]
I'm merging this PR even if the build_and_test CI check does not pass. As we already know:
- MPICH, used in MacOS CI, suffers from a bug that breaks our simulator runs (fixed but not yet released https://github.com/pmodels/mpich/issues/5977)
- Clang 13 on Windows suffers from a bug in its header ```stdatomic.h``` (fixed in LLVM 14 https://github.com/llvm/llvm-project/commit/1ad7de9e92bc2977698e5f6d6493202b50c912d5)
